### PR TITLE
[WIP] Add package: luakit

### PR DESCRIPTION
**Pull request by matthiasbeyer**
**This is WIP**

Package: luakit

It already builds, though when running luakit it fails with the following trace (which is a runtime error):

```
/home/m/.config/luakit/rc.lua:5: module 'lfs' not found:
        no field package.preload['lfs']
        no file './lfs.lua'
        no file '/nix/store/yp6izadskypdvbnpxmv1a0adhibvqnhf-luajit-2.0.3/share/luajit-2.0.3/lfs.lua'
        no file '/usr/local/share/lua/5.1/lfs.lua'
        no file '/usr/local/share/lua/5.1/lfs/init.lua'
        no file '/nix/store/yp6izadskypdvbnpxmv1a0adhibvqnhf-luajit-2.0.3/share/lua/5.1/lfs.lua'
        no file '/nix/store/yp6izadskypdvbnpxmv1a0adhibvqnhf-luajit-2.0.3/share/lua/5.1/lfs/init.lua'
        no file './lib/lfs.lua'
        no file './lib/lfs/init.lua'
        no file './config/lfs.lua'
        no file './config/lfs/init.lua'
        no file '/home/m/.config/luakit/lfs.lua'
        no file '/home/m/.config/luakit/lfs/init.lua'
        no file '/home/m/.nix-profile/etc/xdg/luakit/lfs.lua'
        no file '/home/m/.nix-profile/etc/xdg/luakit/lfs/init.lua'
        no file '/nix/var/nix/profiles/default/etc/xdg/luakit/lfs.lua'
        no file '/nix/var/nix/profiles/default/etc/xdg/luakit/lfs/init.lua'
        no file '/run/current-system/sw/etc/xdg/luakit/lfs.lua'
        no file '/run/current-system/sw/etc/xdg/luakit/lfs/init.lua'
        no file '/nix/store/lkxnbqcz2n5r721w9nyiwk93krzrzrs7-luakit-2012.09.13/share/luakit/lib/lfs.lua'
        no file '/nix/store/lkxnbqcz2n5r721w9nyiwk93krzrzrs7-luakit-2012.09.13/share/luakit/lib/lfs/init.lua'
        no file './lfs.so'
        no file '/usr/local/lib/lua/5.1/lfs.so'
        no file '/nix/store/yp6izadskypdvbnpxmv1a0adhibvqnhf-luajit-2.0.3/lib/lua/5.1/lfs.so'
        no file '/usr/local/lib/lua/5.1/loadall.so'
[    0.018698] E: luakit: main:189: no windows spawned by rc file, exiting
```

My luakit configuration is the default.

@flosse maybe you can help, you packaged a lua package?

### CODE REVIEW

**Comment by lethalman** [comment]
The postInstall doesn't get evaluated because you have overridden installPhase.

**Comment by matthiasbeyer** [comment]
Thanks. Anyways, that does not solve my problem, unfortunately.

**Comment by lethalman** [comment]
Doesn't build for me locally, make sure you have enabled chroot builds:
```
building common/tokenize.h
./build-utils/gentokens.lua common/tokenize.list common/tokenize.h
make: ./build-utils/gentokens.lua: Command not found
make: *** [common/tokenize.h] Error 127
builder for ‘/nix/store/b5v6zxn0px03w8via3vnb3d5hisp3kxv-luakit-2012.09.13.drv’ failed with exit code 2
error: build of ‘/nix/store/b5v6zxn0px03w8via3vnb3d5hisp3kxv-luakit-2012.09.13.drv’ failed
```

**Comment by flosse** [comment]
@matthiasbeyer I'm not a lua dev. I just packed some libs to get prosody running ;-)

**Comment by matthiasbeyer** [comment]
@lethalman How to enable chroot builds?

@flosse I'm not a lua dev either, I asked you because you already packaged stuff... but yeah, never mind.

**Comment by flosse** [comment]
>@flosse I'm not a lua dev either, I asked you because you already packaged stuff... but yeah, never >mind.

I see. I can't promise but maybe I'll have a look at it next week.

**Comment by matthiasbeyer** [comment]
@flosse :+1:

**Comment by bennofs** [comment]
I think I have a derivation for luakit lying around somewhere, I'll look whether I can find it tomorrow

**Comment by bennofs** [comment]
Ah, found it: https://github.com/bennofs/nixpkgs/commit/4830be369dff3265a4be942100dc9a58ece6b9d3

**Comment by matthiasbeyer** [comment]
@bennofs Nice! Would you mind if I integrate this in my patch? I'd add a "Suggested-by:" tag in the commit message.

**Comment by bennofs** [comment]
Not at all, feel free to use it in any way you wish, that's why I posted it in the first place.

**Comment by matthiasbeyer** [comment]
It now tells that:

```bash
$ luakit
(.luakit-wrapped:15157): GLib-GIO-ERROR **: No GSettings schemas are installed on the system
Trace/breakpoint trap
```

But the gsettings-desktop-schemas is in the build inputs... so it seems that luakit cannot find it. Has someone an idea for this?

**Comment by lethalman** [comment]
Missing `$GSETTINGS_SCHEMAS_PATH` in --prefix. Also please collapse --prefix of the same variable.

**Comment by matthiasbeyer** [comment]
> Missing $GSETTINGS_SCHEMAS_PATH in --prefix. 

How to include it properly?

> Also please collapse --prefix of the same variable.

What do you mean by that?

**Comment by lethalman** [comment]
@matthiasbeyer there's plenty of other derivations using $GSETTINGS_SCHEMAS_PATH and also setting XDG_DATA_DIRS.

**Comment by matthiasbeyer** [comment]
ah, okay. I also understand what you mean by "collapsing" now! Thanks!

**Comment by matthiasbeyer** [comment]
Please review.

Before you merge, I'd like to clean up the history and add a "Suggested-by: @bennofs"-like line in the commit message(s), as some parts of the code where ripped from his `add-luakit` branch. (Thanks for that!)

**Comment by matthiasbeyer** [comment]
@bennofs Is it okay for you if I put you in the commit messages with a "Suggested-by: " line including your name and mail address (as the git commit message tags are supposed to be...)?

File: pkgs/applications/networking/browsers/luakit/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{ stdenv
+, fetchurl
```
Comment (lethalman) [inline pkgs/applications/networking/browsers/luakit/default.nix]: Put these on multiple lines, but not one word per line, please. Group them a little.

File: pkgs/applications/networking/browsers/luakit/default.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    maintainers = with maintainers; [ matthiasbeyer ];
+    platforms   = platforms.linux; # I only tested linux
+  };
+
+  src = fetchurl {
+    url = "https://github.com/mason-larobina/luakit/archive/2012.09.13-r1.tar.gz";
+    sha256 = "067g3bp5w20jypc8rx54dpbn3ysbgxzchgpq7sld5yh2z36s1l52";
+  };
+
+  buildInputs = [
```
Comment (lethalman) [inline pkgs/applications/networking/browsers/luakit/default.nix]: Same here, not all those lines please.

File: pkgs/applications/networking/browsers/luakit/default.nix [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    make DEVELOPMENT_PATHS=0 USE_LUAJIT=1 INSTALLDIR=$out DESTDIR=$out PREFIX=$out USE_GTK3=1
+  '';
+
+  installPhase = let
+    luaKitPath = "$out/share/luakit/lib/?/init.lua;$out/share/luakit/lib/?.lua";
+  in ''
+    make DEVELOPMENT_PATHS=0 INSTALLDIR=$out DESTDIR=$out PREFIX=$out USE_GTK3=1 install
+    wrapProgram $out/bin/luakit                                         \
+      --prefix GIO_EXTRA_MODULES : "${glib_networking}/lib/gio/modules" \
+      --prefix XDG_DATA_DIRS : "${gsettings_desktop_schemas}/share"     \
```
Comment (lethalman) [inline pkgs/applications/networking/browsers/luakit/default.nix]: Group those `--prefix XDG_DATA_DIRS` into a single --prefix.

**Comment by bennofs** [comment]
@matthiasbeyer that's ok

**Comment by matthiasbeyer** [comment]
@lethalman Rebased and squashed into one commit. I'm done here, if you have no more annotations for me! :smile:

**Comment by pSub** [comment]
Nitpicking: `description` should not end with a dot.

**Comment by lethalman** [comment]
Travis passed, thanks.

**Comment by lethalman** [comment]
Fixed description in master too.

**Pull request merged**